### PR TITLE
Historical kraken updates

### PR DESCRIPTION
**Pull request by rewong03**
### Description
This addresses #56 and provides an interface to download snapshots for an asset and also to read the asset

### Testing
unit testing and some manual testing

### Impact
standalone, but will be painful for people next quarter if this is wrong lmao

### Other
- since we aren't starting downloads at the beginning of history, the snapshots will be slightly innaccurate
- i realized that i can definitely fetch the data from the api in parallel and i will do that in a different pr

### CODE REVIEW

**Comment by echavemann** [comment]
king shit

File: src/pysrc/adapters/messages.py [hunk 1]
```diff
@@ -31,6 +33,51 @@ def get_bids(self) -> list[tuple[float, float]]:
...
+            self.time,
+            self.market.value,
+            len(self.feedcode),
+            len(bids),
+            len(asks),
+        )
+        return packed_metadata + str.encode(self.feedcode) + bids + asks
+
+    @staticmethod
+    def from_bytes(b: bytes) -> "SnapshotMessage":
```
Comment (echavemann) [inline src/pysrc/adapters/messages.py:51]: isnt thre a way u can do cls for this?

**Review by echavemann — APPROVED** [review]

File: src/pysrc/adapters/kraken/historical/updates/historical_updates_data_client.py
Comment (echavemann) [inline src/pysrc/adapters/kraken/historical/updates/historical_updates_data_client.py:1]: hmm lwk I dont love the download and reading APIs being together

Comment (rewong03) [reply src/pysrc/adapters/kraken/historical/updates/historical_updates_data_client.py:1]: kind of agree i think it makes more sense for there to be trade/snapshot serializer/deserializer classes, someone else can take that and refactor

File: src/pysrc/adapters/messages.py (hunk 1 above)
Comment (rewong03) [reply src/pysrc/adapters/messages.py:51]: are you talking about the "SnapshotMessage" type

**Pull request merged**